prov/efa/src/rdm/efa_rdm_peer.h

```diff
@@ -128,6 +128,47 @@ bool efa_both_support_rdma_read(struct efa_rdm_ep *ep, struct efa_rdm_peer *peer
 	       (peer->is_self || efa_rdm_peer_support_rdma_read(peer));
 }
 
+/**
+ * @brief Determine interoperability for RDMA read between peers
+ *
+ * RDMA read is currently not interoperable between all EFA platforms; older
+ * platforms cannot fully interoperate with newer platforms. The platform
+ * version can be inferred from ibv_device_attr.vendor_part_id (see
+ * ibv_query_device(3)).
+ *
+ * In this context, version 0xEFA0 is considered "older", while versions 0xEFA1
+ * or 0xEFA2 are considered "newer"
+ *
+ * @todo Either:
+ * - Refactor this logic if the interoperability is ever reported
+ *   directly via firmware or rdma-core
+ * - Remove altogether if RDMA read becomes interoperable between all EFA
+ *   platforms
+ *
+ * @param[in]	ep	Endpoint for communication with peer
+ * @param[in]	peer	An EFA peer
+ *
+ * @return true if peers can interoperate via RDMA read; false otherwise
+ */
+static inline
+bool efa_rdm_interop_rdma_read(struct efa_rdm_ep *ep, struct efa_rdm_peer *peer)
+{
+	bool rdma_read_support = efa_both_support_rdma_read(ep, peer);
+	uint32_t ep_dev_ver = efa_rdm_ep_domain(ep)->device->ibv_attr.vendor_part_id,
+		 peer_dev_ver = peer->device_version;
+
+	switch (ep_dev_ver) {
+	case 0xEFA0:
+		return rdma_read_support && peer_dev_ver == 0xEFA0;
+	case 0xEFA1:
+	case 0xEFA2:
```

Comment:
We should only care 0xefa0 and NOT 0xefa0 cases here. No need to restrict the version id within the [0xefa0, 0xefa2] range. See Brian's comment in https://github.com/ofiwg/libfabric/pull/8186#discussion_r1008588620

Reply:
Lol, I chose to explicitly compare against `0xEFA[0-2]` due to his other comment further down: https://github.com/ofiwg/libfabric/pull/8186#discussion_r1009783855

I don't feel strongly either way.

Reply:
I think they don't conflict. Brian wants to have a explicit check for 0xefa0 and 0xefa1 because we don't want to enable rdma-read for them by default. He doesn't care the others because we are enabling them by default.

Here the case is similar. We just don't allow rdma-read traffic between AL7 and non-AL7.